src/html/parse-frontmatter.js

```diff
@@ -0,0 +1,206 @@
+const yaml = require('js-yaml');
+
+/**
+ * Given an mdast tree and it's text representation, this finds all
+ * the frontmatter in it.
+ *
+ * Frontmatter looks like this:
+ *
+ * ```
+ * ---
+ * Foo: bar
+ * ---
+ * ```
+ *
+ * The frontmatter is delimited by a `thematicBreak` made from three
+ * dashes in the mdast and contains a yaml-encoded payload.
+ *
+ * Not any yaml and any thematicBreak is accepted; in order to avoid
+ * false positives, the following restrictions apply:
+ *
+ * - There must be the start or end of the document or an empty line (`\n\n`)
+ *   before and after the frontmatter block
+ * - There may be no empty line within the frontmatter block; not even lines
```

Comment:
That might be too strict. Can we losen "within" to "at the start or end"?

Reply:
I've considered over the last couple days, and tbh I don't really know if this can be done safely…
We can introduce explicit frontmatter `---yaml` which should make false positives impossible and then also reintroduce empty lines in frontmatter…